BeagleBone_black/src/sensors/fake_imu.cpp

```diff
@@ -101,51 +105,106 @@ void FakeImu::readDataFromFile(std::string acc_file_path, std::string gyr_file_p
 {
   std::ifstream file;
   uint32_t timestamp;
-  NavigationVector temp;
+  NavigationVector value, noise;
 
   file.open(acc_file_path);
-
-  file >> acc_noise;
   while (file.is_open() && !file.eof()) {
-    file >> timestamp >> temp[0] >> temp[1] >> temp[2];
-    acc_val_read.push_back(DataPoint<NavigationVector>(timestamp, addNoiseToData(temp, acc_noise)));
+    file >> timestamp >> value[0] >> value[1] >> value[2]
+         >> noise[0] >> noise[1] >> noise[2];
+    acc_val_read.push_back(DataPoint<NavigationVector>(timestamp, addNoiseToData(value, noise)));
   }
 
   if (file.is_open())
     file.close();
 
   file.open(gyr_file_path);
-  file >> gyr_noise;
   while (file.is_open() && !file.eof()) {
-    file >> timestamp >> temp[0] >> temp[1] >> temp[2];
-    gyr_val_read.push_back(DataPoint<NavigationVector>(timestamp, addNoiseToData(temp, gyr_noise)));
+    file >> timestamp >> value[0] >> value[1] >> value[2]
+         >> noise[0] >> noise[1] >> noise[2];
+    gyr_val_read.push_back(DataPoint<NavigationVector>(timestamp, addNoiseToData(value, noise)));
   }
+
+  if (file.is_open())
+    file.close();
 }
 
 bool FakeImu::accCheckTime()
 {
-    high_resolution_clock::time_point now = high_resolution_clock::now();
-    duration<double> time_span = duration_cast<duration<double>>(now - imu_ref_time);
+  high_resolution_clock::time_point now = high_resolution_clock::now();
+  microseconds time_span = duration_cast<microseconds>(now - imu_ref_time);
 
-    if (time_span.count() < 0.000250*acc_count) {
-        return false;
-    }
+  if (unsigned(time_span.count()) < kAccTimeInterval*acc_count) {
+    return false;
+  }
 
-    acc_count++;
-    return true;
+  acc_count = time_span.count()/kAccTimeInterval + 1;
+  return true;
 }
 
 bool FakeImu::gyrCheckTime()
 {
-    high_resolution_clock::time_point now = high_resolution_clock::now();
-    duration<double> time_span = duration_cast<duration<double>>(now - imu_ref_time);
+  high_resolution_clock::time_point now = high_resolution_clock::now();
+  microseconds time_span = duration_cast<microseconds>(now - imu_ref_time);
+
+  if (unsigned(time_span.count()) < kGyrTimeInterval*gyr_count) {
+    return false;
+  }
+
+  gyr_count = time_span.count()/kGyrTimeInterval + 1;
+  return true;
+}
 
-    if (time_span.count() < 0.000125*gyr_count) {
+bool FakeImu::checkFile(std::string acc_file_path, std::string gyr_file_path)
```

Comment:
Move the checking functionality to `readDataFromFile()` or otherwise make sure you only read each file once. The files can potentially have millions of lines so reading them twice is extremely wasteful.